### PR TITLE
Add ability to filter the collection by numerical columns

### PR DESCRIPTION
**Pull request by Dakes**
Since someone in the forum was asking how to filter playlists, I thought it would be useful to have a tooltip there, like the collection filter has. 
This person also named playcount as an example, which I thought would be pretty useful to have, so I added it together with the skipcount to the playlist filtering. 

Now for the big addition: I added filtering of numerical colums to the collection. So that you can filter it like: `rating:>3`
I tested it a while for the numerical cols: year, length, samplerate, bitdepth, bitrate, rating, playcount, skipcount and it seems to work great, also when combined with fts search terms. 
Though I am a bit wary about the comment, regarding performance and FilterModes. I don't really understand what is going on there, but I don't think it is relevant to what I added. 

I also created a new file `src/utilities/searchparserutils.cpp` to house common code for search parsing, used by the playlist and collection (and in the future maybe smart playlists? But I haven't looked at that code yet.)

I hope these changes are fine and not too intrusive. 

And one more question: since the changed/added tooltips, they require new translations, it is probably necessary to update the translation files. But I have not found documentation on how to do that. Could you point me in the right direction there?

### CODE REVIEW

File: src/collection/collectionquery.cpp
Comment (jonaski) [inline src/collection/collectionquery.cpp:149]: Move { to the previous line
Add one empty line after { and before } similar to the other code.

File: src/collection/collectionquery.cpp
Comment (jonaski) [inline src/collection/collectionquery.cpp:154]: ident 2 spaces

File: src/collection/collectionquery.cpp
Comment (jonaski) [inline src/collection/collectionquery.cpp:162]: New line before }

File: src/collection/collectionquery.cpp
Comment (jonaski) [inline src/collection/collectionquery.cpp:232]: move { to previous line

File: src/playlist/playlistfilter.h
Comment (jonaski) [inline src/playlist/playlistfilter.h:55]: Add a const accessor function instead of moving this to public

File: src/utilities/searchparserutils.cpp
Comment (jonaski) [inline src/utilities/searchparserutils.cpp:43]: ParseSearchTime

File: src/utilities/searchparserutils.cpp
Comment (jonaski) [inline src/utilities/searchparserutils.cpp:78]: ParseSearchRating

**Review by jonaski — CHANGES_REQUESTED** [review]
General code style: Add one empty line after { and before } for functions, similar to the other code.

This is probably fine, but beware that I'm trying to move away from SQL based collection search to a model based search, see #729, which means the collection changes might have to be redone later.

**Comment by jonaski** [comment]
For translations see https://wiki.strawberrymusicplayer.org/wiki/How_to_translate
But you need to wait until this is merged and @strawbsbot has committed a translations update.

File: src/playlist/playlistfilter.h
Comment (jonaski) [inline src/playlist/playlistfilter.h:53]: Why the extra semi-colon after }?

File: src/playlist/playlistfilter.h
Comment (jonaski) [inline src/playlist/playlistfilter.h:54]: Remove ; at the end

File: src/collection/collectionquery.cpp
Comment (jonaski) [inline src/collection/collectionquery.cpp:83]: Remove extra space before //

**Review by jonaski — CHANGES_REQUESTED** [review]

File: src/collection/collectionquery.cpp
Comment (jonaski) [inline src/collection/collectionquery.cpp:151]:
```suggestion
  static QRegularExpression rxOp("^(=|<[>=]?|>=?|!=)");
```

File: src/playlist/playlistfilterparser.cpp
Comment (jonaski) [inline src/playlist/playlistfilterparser.cpp:522]: parsed_search

File: src/collection/collectionquery.cpp
Comment (jonaski) [inline src/collection/collectionquery.cpp:216]: parsed_rating

**Review by jonaski — CHANGES_REQUESTED** [review]

File: src/playlist/playlistfilter.h
Comment (Dakes) [reply src/playlist/playlistfilter.h:53]: I probably assumed I deleted one by mistake, because I am not used to seeing inline function definitions. 
I will try to be more thorough from now on. Also regarding the general style.

**Review by jonaski — CHANGES_REQUESTED** [review]

File: src/collection/collectionquery.cpp
Comment (Dakes) [reply src/collection/collectionquery.cpp:150]: I changed it so, that it initializes op with "=", to return = by default. In exchange the empty check after line 88 was removed

**Pull request merged**
